src/main/java/com/example/petbuddybackend/service/care/state/CareStateMachine.java

```diff
@@ -32,14 +42,15 @@ public void transitionToEditCare(Care care) {
         }
 
         if(!caretakerStatus.equals(CareStatus.PENDING)) {
-            throw new StateTransitionException(new Transition(Role.CARETAKER, caretakerStatus, CareStatus.PENDING));
+            throw new StateTransitionException(new RoleTransition(Role.CARETAKER, caretakerStatus, CareStatus.PENDING));
         }
 
         if(!(clientStatus.equals(CareStatus.ACCEPTED) || clientStatus.equals(CareStatus.PENDING))) {
-            throw new StateTransitionException(new Transition(Role.CLIENT, clientStatus, CareStatus.PENDING));
+            throw new StateTransitionException(new RoleTransition(Role.CLIENT, clientStatus, CareStatus.PENDING));
         }
 
-        setBothStatuses(care, CareStatus.PENDING);
+        care.setCaretakerStatus(CareStatus.ACCEPTED);
+        care.setClientStatus(CareStatus.PENDING);
     }
 
     private void initClientTransitions(TransitionManager transitionManager) {
```

Comment:
`transitionManager.addTransition(Role.CLIENT, CareStatus.PENDING, CareStatus.ACCEPTED, this::acceptAsClient);`
bdziemy sprawdzali warunek, zeby opiekun mial wczesniej ACCEPTED zanim klient moze zrobic ACCEPTED?

Reply:
`transitionManager.addTransition(Role.CLIENT, CareStatus.ACCEPTED, CareStatus.CANCELLED, this::setBothStatuses);`

w teorii jesli opiekun ma status accepted i klient tez to statusy z automatu leca na AWAITING_PAYMENT, ale nwm czy tez nie dodalbym warunku, ze klient moze zmienic z ACCEPTED na PENDING jesli opiekun nie ma tego na ACCEPTED

Reply:
`transitionManager.addTransition(Role.CARETAKER, CareStatus.ACCEPTED, CareStatus.CANCELLED, this::setBothStatuses);`

przy opiekunie taka sama sytuacja jak wyzej

Reply:
> transitionManager.addTransition(Role.CLIENT, CareStatus.PENDING, CareStatus.ACCEPTED, this::acceptAsClient);
bdziemy sprawdzali warunek, zeby opiekun mial wczesniej ACCEPTED zanim klient moze zrobic ACCEPTED?

Imo rozluźniłbym troche te statusy. Nie widzę problemu z tym żeby można było zawsze akceptować jeśli jest `PENDING`. Dodając to ograniczenie zakładamy, że system się nie będzie mylił i nie doprowadzimy do takiej sytuacji, gdzie i klient i opiekun mają `PENDING`. 

Ogólnie tak, możemy założyć że taka sytuacja może być błędna, ale imo mniejszym problemem będzie jeżeli obaj użytkownicy będą `PENDING` i będą mogli zaakceptować niż jak obaj będą `PENDING` i nie będą mogli z tego stanu wyjść. 

Chyba że czegoś nie widzę w tym przykładzie i faktycznie to by coś nam psuło

Reply:
`

transitionManager.addTransition(Role.CLIENT, CareStatus.ACCEPTED, CareStatus.CANCELLED, this::setBothStatuses);

w teorii jesli opiekun ma status accepted i klient tez to statusy z automatu leca na AWAITING_PAYMENT, ale nwm czy tez nie dodalbym warunku, ze klient moze zmienic z ACCEPTED na PENDING jesli opiekun nie ma tego na ACCEPTED
`

W sensie mógłbym dać ACCEPTED -> PENDING ale nie sprawdzałbym wtedy czy druga strona ma ACCEPT bo imo od tego jest ten AWAITING_PAYMENT żeby nie można było z tego "wrócić"

Reply:
w sumie mozna rozluznic chyba z PENDING do ACCEPT, bo nawet jak oboje beda mieli PENDING to w sumie nic nie popsuje, a nawet chyab zabardzo nie da sie zrobic takiej sytuacji

Reply:
a co do drugiego, to tez mozna, bo jak oboje maja ACCEPTED to skacze na AWAITING_PAYMENT, wiec tez kazda sytuacja ACCEPTED - CANCELLED bedzie tylko jesli jedno z nich mialo ACCEPTED